gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -78,6 +84,56 @@
     this.showUnsavedChangesDialogEvents = new SingleLiveEvent<>();
     this.showErrorDialogEvents = new SingleLiveEvent<>();
     this.authManager = authenticationManager;
+    this.editRecordRequests = PublishSubject.create();
+    this.recordSaveRequests = PublishSubject.create();
+
+    disposeOnClear(
+        recordSaveRequests
+            .switchMap(this::saveRecord)
+            .subscribe(record::setValue, this::onSaveRecordError));
```

Comment:
Don't worry about it here, but we should change this to use the `doOnError` handling instead of handling the error in the subscription.

Reply:
I'm pretty sure this is wrong, actually; either the Resource should wrap the error, or it should be handled on the inner stream. I'll add a TODO.